### ISSUE TITLE
SMTPMailer does not reconnect to SMTP server

### ISSUE THREAD

**Issue by rbu** [zopefoundation/zope.sendmail#1]
```zope.sendmail.mailer.SMTPMailer``` does not reestablish an SMTP connection after sending the first mail.

```SMTPMailer``` in previous versions of zope.sendmail (up until 3.7.5) would always establish an new SMTP connection when ```send``` was being called. Its very own ```QueueProcessorThread``` and other projects may rely on the fact that one ```SMTPMailer``` instance can be reused (immediately or after some time) to send multiple mails.
However, since 4.0.0 the object caches the ```self.connection``` (an  ```SMTP``` instance). At the end of ```send()``` it will call ```self.connection.quit()``` and/or ```self.connection.close()```, making the connection unusable. Similar code is in ```abort()```. In all of these cases, ```self.connection``` should be set to ```None```, or the usage of ```self.connection``` should be reverted to a local variable ```connection``` that is returned by ```self.vote()```.

**Comment by jamadden** [comment]
I agree, this is definitely a bug. `SMTPMailer.send()` unconditionally calls `connection.quit()` which itself unconditionally calls `connection.close()`, so the SMTPMailer is useless after one message has been sent. Given that SMTPMailers are registered as utilities, that seems wrong. And given that the `zope-sendmail` command (from `queue.py`) will try to send messages in a loop using the same SMTPMailer object, that *also* seems wrong.

We have a good base for adding tests and fixing this with #11.
